From 39caebddb9b8f15d1d4fc8471f2a8ba7f10b5fdd Mon Sep 17 00:00:00 2001
From: "Thulasimani,Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Tue, 29 Mar 2016 11:24:06 +0530
Subject: [PATCH 15/19] FOR_UPSTREAM [VPG]: drm/i915: move
 pm_qos_update_request to dp detect

pm_qos_update_request calls take upto a few milliseconds on CHT HR
(and i assume on any board in battery mode). Having it as part of
last minute request before performing dpcd requests results in
delaying each dpcd request and thus slowing the whole enable
sequence. Hence it beter to keep them in detect sequence
itself to avoid making this call multiple times.

Change-Id: Ibd5183fff6f962482b210580d90403db0f34ecc9
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-30099
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Reviewed-on: https://android.intel.com:443/498190
---
 drivers/gpu/drm/i915/intel_dp.c | 17 +++++++----------
 1 file changed, 7 insertions(+), 10 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 0d33c58..a4652c9 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -728,12 +728,6 @@ intel_dp_aux_ch(struct intel_dp *intel_dp,
 	if (is_edp(intel_dp) && !edp_have_panel_power(intel_dp))
 		vdd = _edp_panel_vdd_on(intel_dp);
 
-	/* dp aux is extremely sensitive to irq latency, hence request the
-	 * lowest possible wakeup latency and so prevent the cpu from going into
-	 * deep sleep states.
-	 */
-	pm_qos_update_request(&dev_priv->pm_qos, 0);
-
 	intel_dp_check_edp(intel_dp);
 
 	if (IS_CHERRYVIEW(dev))
@@ -843,7 +837,6 @@ intel_dp_aux_ch(struct intel_dp *intel_dp,
 
 	ret = recv_bytes;
 out:
-	pm_qos_update_request(&dev_priv->pm_qos, PM_QOS_DEFAULT_VALUE);
 	intel_display_power_put(dev_priv, pipe);
 
 	if (vdd)
@@ -4882,6 +4875,7 @@ intel_dp_detect(struct drm_connector *connector, bool force)
 		intel_dp->aux.i2c_defer_count = 0;
 
 		intel_dp->has_audio =  false;
+		pm_qos_update_request(&dev_priv->pm_qos, PM_QOS_DEFAULT_VALUE);
 		goto out;
 	}
 
@@ -4939,9 +4933,12 @@ intel_dp_detect(struct drm_connector *connector, bool force)
 		chv_upfront_link_train(dev, intel_dp, intel_crtc);
 	}
 
-	/* if simulation was in progress clear the flag */
-	if (dev_priv->simulate_dp_in_progress & intel_encoder->hpd_pin)
-		dev_priv->simulate_dp_in_progress &= ~(intel_encoder->hpd_pin);
+	/*
+	 * dp aux is extremely sensitive to irq latency, hence request the
+	 * lowest possible wakeup latency and so prevent the cpu from going into
+	 * deep sleep states.
+	 */
+	pm_qos_update_request(&dev_priv->pm_qos, 0);
 
 out:
 #ifdef CONFIG_SUPPORT_LPDMA_HDMI_AUDIO
-- 
1.9.1

